src/algebra/big_operators/basic.lean

```diff
@@ -1269,6 +1269,26 @@ lemma prod_pow_boole [decidable_eq α] (s : finset α) (f : α → β) (a : α)
   (∏ x in s, (f x)^(ite (a = x) 1 0)) = ite (a ∈ s) (f a) 1 :=
 by simp
 
+@[to_additive]
+lemma prod_ite_eq_extract  [decidable_eq α]
+ (s : finset α) {b : α} (hb : b ∈ s) (f : α → β) :
+  ∏ n in s, f n = f b * ∏ n in s, ite (n = b) 1 (f n) :=
+begin
+  simp only [finset.prod_ite, finset.filter_ne', finset.prod_const_one, one_mul],
+  rw mul_comm,
+  exact (prod_erase_mul _ _ hb).symm,
+end
+
+@[to_additive]
+lemma prod_update_eq [decidable_eq α]
+ (s : finset α) {a : α} (b : β) (ha : a ∈ s)  (f : α → β) :
+  f a * (∏ n in s, (function.update f a b) n)  = b * ∏ n in s, (f n):=
```

Comment:
```suggestion
 (s : finset α) {a : α} (b : β) (ha : a ∈ s) (f : α → β) :
  f a * (∏ n in s, function.update f a b n) = b * ∏ n in s, f n :=
```

Reply:
I removed this lemma as I hadn't seen the `sum_update_of_mem`. Altrhough I can re-add it if you think its worth having.